Source/WebCore/rendering/LayerOverlapMap.cpp

```diff
@@ -314,6 +327,35 @@ void LayerOverlapMap::popCompositingContainer(const RenderLayer& layer)
     m_overlapStack.removeLast();
 }
 
+void LayerOverlapMap::pushSpeculativeCompositingContainer(const RenderLayer& layer)
+{
+    // Create a duplicate copy of the overlap stack, push the container to one,
+    // and make all future add calls apply to both stacks.  
+    // If we end up needing the compositing container, then we throw away the
+    // backup, otherwise we restore the primary stack using the backup.
+    // If we already have a backup (and we've recursed into this), then just force
+    // a real compositing container on the outer, since otherwise we'd have to start
+    // tracking four possible outcomes.
+    confirmSpeculativeCompositingContainer();
+    for (auto& container : m_overlapStack)
+        m_backupOverlapStack.append(makeUnique<OverlapMapContainer>(*container));
+    m_overlapStack.append(makeUnique<OverlapMapContainer>(m_rootLayer, layer));
+}
+
+void LayerOverlapMap::confirmSpeculativeCompositingContainer()
+{
+    m_backupOverlapStack.clear();
+}
+
+bool LayerOverlapMap::maybePopSpeculativeCompositingContainer()
```

Comment:
You're using both "backup" and "speculative" term here, and you should choose just one of them (e.g. `m_speculativeOverlapStack`).

Reply:
The 'backup' stack is what would have happened if we didn't push the speculative container, so the two terms are both meaningful.

What I can do though is swap the meanings, so that the explicit extra stack is the one containing the speculative change.